packages/web-app-external/src/App.vue

```diff
@@ -132,12 +149,49 @@ export default defineComponent({
       }
     }).restartable()
 
+    // switch to write mode when edit is clicked
+    const catchClickMicrosoftEdit = (event) => {
+      try {
+        if (JSON.parse(event.data)?.MessageId === 'UI_Edit') {
+          loadAppUrl.perform('write')
+        }
+      } catch (e) {}
+    }
+    watch(
+      applicationName,
+      (newAppName, oldAppName) => {
+        console.log('appNames', newAppName, oldAppName)
+        if (newAppName === 'Office365' && newAppName !== oldAppName) {
+          window.addEventListener('message', catchClickMicrosoftEdit)
+        } else {
+          window.removeEventListener('message', catchClickMicrosoftEdit)
+        }
+      },
+      {
+        immediate: true
+      }
+    )
```

Comment:
It's not so nice to have this kind of specific handling here, but handling this as an extension seems overkill right now... it would be nice if all applications would emit the same kind of event, but that might be impossible especially with proprietary third party applications...